### DISCUSSION TITLE
Implement hobby's algorithm for slider creation

### DISCUSSION THREAD

**Discussion by verdiur** [ppy/osu#27191] [Ideas]
Taken from Jake Low's blog post : 

> Hobby’s algorithm is an alternative technique for choosing handle positions given a sequence of knots, in order to produce a Bézier spline. Instead of requiring continuity of true curvature, as with the natural cubic spline above, Hobby’s algorithm produces splines that have continuity of mock curvature (a first-order approximation of true curvature).

I think the implementation of Hobby's algorithm for slider creation would ~~complete the hand-drawn slider tool nicely~~ make it more user friendly to place sliders by manually placing anchors; unless such a method for natural handle positioning is already implemented? I'm not too familiar with the current state of lazer mapping.

Here are a few resources:
[Blog post by Jake Low](http://www.jakelow.com/blog/hobby-curves)
[John D. Hobby's paper](https://link.springer.com/content/pdf/10.1007/BF02187690.pdf)

**Comment by bdach** [comment]
The hand-drawing thing already attempts to optimise the path created to fit the slider shape to the cursor movements _while limiting the number of anchors output_. I'm not sure the algorithm you propose is appropriate for hand-drawing because the limiting of anchor count is a _requirement_.

**Reply by verdiur** [reply]
Sorry for not clarifying properly. I'm not proposing to use this algorithm for hand-drawing but rather as another way of making sliders by placing anchors like in stable. I know the "placing anchors" method is already implemented but I think an option to use this algorithm with cubics could prove to be useful in many cases, if it isn't already implemented.
